sdks/router-sdk/src/entities/mixedRoute/trade.ts

```diff
@@ -357,16 +365,20 @@ export class MixedRouteTrade<TInput extends Currency, TOutput extends Currency,
     )
 
     const numPools = routes.map(({ route }) => route.pools.length).reduce((total, cur) => total + cur, 0)
-    const poolAddressSet = new Set<string>()
+    const poolIdentifierSet = new Set<string>()
     for (const { route } of routes) {
       for (const pool of route.pools) {
-        pool instanceof Pool
-          ? poolAddressSet.add(Pool.getAddress(pool.token0, pool.token1, pool.fee))
-          : poolAddressSet.add(Pair.getAddress(pool.token0, pool.token1))
+        if (pool instanceof Pair) {
+          poolIdentifierSet.add(Pair.getAddress(pool.token0, pool.token1))
+        } else if (pool instanceof V3Pool) {
+          poolIdentifierSet.add(V3Pool.getAddress(pool.token0, pool.token1, pool.fee))
+        } else if (pool instanceof V4Pool) {
+          poolIdentifierSet.add(pool.poolId)
+        }
```

Comment:
I think we should throw in an else branch here for unknown pool types, would help catch mismatched package issues

Reply:
I can throw this in, althoughhh I have noticed we have these kinds of checks exist in places, but realize they're redundant because every function that accepts pools has a `TPool = Pair | V3Pool | V4Pool`constraint for inputs